linera-base/src/abi.rs

```diff
@@ -70,3 +72,24 @@ where
     type Query = <<A as WithServiceAbi>::Abi as ServiceAbi>::Query;
     type QueryResponse = <<A as WithServiceAbi>::Abi as ServiceAbi>::QueryResponse;
 }
+
+/// The type of a query for an oracle.
+pub trait OracleQuery: Send + Sync + Debug + 'static {
```

Comment:
Following the coding pattern of `ContractAbi`, there should be two associated types: queries and responses.

Reply:
I wanted the user to be able to call `query_oracle` without specifying a generic argument explicitly, because that's already determined by the query itself.

But I guess we could have the `OracleQuery` trait have an associated type as well that links back to the `Oracle` trait?